### PR TITLE
:fire: add config to enable splitting by comma in parsers

### LINKED ISSUE #2557: 🐛 [Bug]: BodyParser will fail to parse correctly into the slices of the struct if input data contains comma
### Bug Description

In an HTML form, if there are multiple input elements with same name and the user enters content containing **commas**(,), thenBodyParser() function will fail to parse correctly into the slices member of the struct

```html
<html>
      <head>
      </head>
      <body>
        <form method="post" action="/">
          <input name='todo'  type='text' value='a,b'>
          <input name='todo' type='text' value='c'>
          <input type='submit' value='submit'>
      </form>
      </body>
  </html>
```
```go
  
  func (c *Test) Test(ctx *fiber.Ctx) error {
	  type MyForm struct {
		  Todo []string `form:"todo"`
	  }
	  
	  var frm MyForm
	  ctx.BodyParser(&frm)
	  
          //incorrect member size 3
          // frm.Todo[0] = "a"      frm.Todo[1]="b"   frm.Todo[2]="c"
        
         // correct members should be:  frm.Todo[0]="a,b"      frm.Todo[1]="c"

	  fmt.Println(len(frm.Todo))
  }
```

### How to Reproduce

1. Go to '....'
3. Do '....'
4. See '....'

### Expected Behavior

The number of elements in the slice member parsed into the struct must be the same as the number of input elements in the HTML form, and their content must also match.

### Fiber Version

v2.48.0

### Code Snippet (optional)

```html
<html>
      <head>
      </head>
      <body>
        <form method="post" action="/">
          <input name='todo'  type='text' value='a,b'>
          <input name='todo' type='text' value='c'>
          <input type='submit' value='submit'>
      </form>
      </body>
  </html>
```

```go
package main

import "github.com/gofiber/fiber/v2"
import "log"

func main() {
  app := fiber.New()

  // Steps to reproduce
	app.Post("/",func(ctx *fiber.Ctx) error {
		type MyForm struct {
			Todo []string `form:"todo"`
		}

		var frm MyForm
		ctx.BodyParser(&frm)

	    //incorrect member size 3
          // frm.Todo[0] = "a"      frm.Todo[1]="b"   frm.Todo[2]="c"
        
         // correct members should be:  frm.Todo[0]="a,b"      frm.Todo[1]="c"
		return nil
	})
  log.Fatal(app.Listen(":3000"))
}
```

### Checklist:

- [X] I agree to follow Fiber's [Code of Conduct](https://github.com/gofiber/fiber/blob/master/.github/CODE_OF_CONDUCT.md).
- [X] I have checked for existing issues that describe my problem prior to opening this one.
- [X] I understand that improperly formatted bug reports may be closed without explanation.

### PR DESCRIPTION
**Pull request by efectn**
## Description

This PR adds a config field to enable the feature that was implemented in https://github.com/gofiber/fiber/pull/817. It's disabled by default since it's not common pattern in web frameworks. Also added escape char support to avoid splitting some fields when we want.
Fixes #2557 

Benchmarks Before:
```
Benchmark_Ctx_BodyParser_JSON
Benchmark_Ctx_BodyParser_JSON-16                                                     	 2245416	       533.9 ns/op	     240 B/op	       6 allocs/op
Benchmark_Ctx_BodyParser_XML
Benchmark_Ctx_BodyParser_XML-16                                                      	  626001	      1902 ns/op	    1160 B/op	      24 allocs/op
Benchmark_Ctx_BodyParser_Form
Benchmark_Ctx_BodyParser_Form-16                                                     	 1261950	       953.7 ns/op	     360 B/op	      13 allocs/op
Benchmark_Ctx_BodyParser_MultipartForm
Benchmark_Ctx_BodyParser_MultipartForm-16                                            	 1412667	       836.6 ns/op	     328 B/op	      12 allocs/op
Benchmark_Ctx_QueryParser
Benchmark_Ctx_QueryParser-16                                                         	  461858	      2507 ns/op	     857 B/op	      38 allocs/op
Benchmark_Ctx_parseQuery
Benchmark_Ctx_parseQuery-16                                                          	  550956	      2174 ns/op	     753 B/op	      29 allocs/op
Benchmark_Ctx_QueryParser_Comma
Benchmark_Ctx_QueryParser_Comma-16                                                   	  424041	      2798 ns/op	     929 B/op	      44 allocs/op
Benchmark_Ctx_ReqHeaderParser
Benchmark_Ctx_ReqHeaderParser-16                                                     	  433765	      2757 ns/op	     929 B/op	      44 allocs/op
```
Benchmarks After:
```
Benchmark_Ctx_Body_With_Compression
Benchmark_Ctx_Body_With_Compression-16                                               	 2044170	       588.4 ns/op
Benchmark_Ctx_BodyParser_JSON
Benchmark_Ctx_BodyParser_JSON-16                                                     	 2250416	       536.8 ns/op	     240 B/op	       6 allocs/op
Benchmark_Ctx_BodyParser_XML
Benchmark_Ctx_BodyParser_XML-16                                                      	  618874	      1897 ns/op	    1160 B/op	      24 allocs/op
Benchmark_Ctx_BodyParser_Form
Benchmark_Ctx_BodyParser_Form-16                                                     	 1239142	       968.6 ns/op	     360 B/op	      13 allocs/op
Benchmark_Ctx_BodyParser_MultipartForm
Benchmark_Ctx_BodyParser_MultipartForm-16                                            	 1432702	       851.8 ns/op	     328 B/op	      12 allocs/op
Benchmark_Ctx_QueryParser
Benchmark_Ctx_QueryParser-16                                                         	  459550	      2566 ns/op	     856 B/op	      38 allocs/op
Benchmark_Ctx_parseQuery
Benchmark_Ctx_parseQuery-16                                                          	  538406	      2205 ns/op	     753 B/op	      29 allocs/op
Benchmark_Ctx_QueryParser_Comma
Benchmark_Ctx_QueryParser_Comma-16                                                   	  403694	      2885 ns/op	     945 B/op	      45 allocs/op
Benchmark_Ctx_ReqHeaderParser
Benchmark_Ctx_ReqHeaderParser-16                                                     	  403273	      2857 ns/op	     945 B/op	      45 allocs/op
```
## Type of change
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Checklist:

- [x] For new functionalities I follow the inspiration of the express js framework and built them similar in usage
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation - /docs/ directory for https://docs.gofiber.io/
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I tried to make my code as fast as possible with as few allocations as possible

## Commit formatting:
Use emojis on commit messages so it provides an easy way of identifying the purpose or intention of a commit. Check out the emoji cheatsheet here: https://gitmoji.carloscuesta.me/

### CODE REVIEW

File: app.go [hunk 1]
```diff
@@ -390,6 +390,15 @@ type Config struct {
 	//
 	// Optional. Default: DefaultMethods
 	RequestMethods []string
+
+	// EnableSplittingOnParsers splits the query/body/header parameters by comma when it's true.
+	// For example, you can use it to parse multiple values from a query parameter like this:
+	//   /api?foo=bar,baz == foo[]=bar&foo[]=baz
+	// If there is an escape character before the comma, parsers won't split it like the example below:
```
Comment (efectn) [inline app.go:397]: Do we need to support escape chars? It has minor performance penalty (1 alloc)

**Review by gaby — APPROVED** [review]
Besides the golangci-lint, lgtm

**Review by ReneWerner87 — CHANGES_REQUESTED** [review]
I will look a second time because of the escaping

File: app.go (hunk 1 above)
Comment (ReneWerner87) [reply app.go:397]: no we do not have to -> no escape support

Comment (ReneWerner87) [reply app.go:397]: but maybe we should already now use a struct for parser configurations for possible later configurations ?

ParserOptions.EnableSplitting

Comment (efectn) [reply app.go]: > but maybe we should already now use a struct for parser configurations for possible later configurations ?
> 
> ParserOptions.EnableSplitting

It inludes config fields related to schema decoder. Should we do it?

Comment (ReneWerner87) [reply app.go]: Sounds good

**Review by ReneWerner87 — APPROVED** [review]

**Pull request merged**